frontend/src/queries/nodes/DataVisualization/DataVisualization.tsx

```diff
@@ -184,6 +188,29 @@ function InternalDataTableVisualization(props: DataTableVisualizationProps): JSX
         visualizationType === ChartDisplayType.ActionsStackedBar
     ) {
         component = <LineGraph />
+    } else if (visualizationType === ChartDisplayType.ActionsPie) {
+        const pieData: GraphDataset[] =
+            yData && xData?.column
+                ? [
+                      {
+                          id: 0,
+                          labels: yData[0]?.data.map((_, i) => xData.data[i] || ''),
+                          data: yData[0]?.data || [],
+                          backgroundColor: yData[0]?.data.map((_, i) => getSeriesColor(i)),
+                          borderColor: yData[0]?.data.map((_, i) => getSeriesColor(i)),
+                      },
+                  ]
+                : []
```

Comment:
From your screenshot, it looks like the pie chart renders with no x-axis series selected and only y-series. This UI us confusing from a UX point of view. X/Y labels don't make sense in a pie chart (even though they're effectively the same). Your screenshot also suggests that multiple y-series can be selected, but we're only using `yData[0]` here?